test_runner/regress/test_ondemand_download.py

```diff
@@ -357,8 +357,23 @@ def get_resident_physical_size():
             tenant_id, timeline_id, "pageserver_resident_physical_size"
         )
 
+    # Shut down safekeepers before starting the pageserver.
+    # If we don't, the tenant's walreceiver handler will trigger the
+    # the logical size computation task, and that downloads layes,
+    # which makes our assertions on size fail.
```

Comment:
```suggestion
```

This conflicts with the fact that we do cause logical size calculation right after with timeline_detail but that is obscured behind `get_api_current_physical_size`.